core/src/mindustry/world/blocks/experimental/BlockLoader.java

```diff
@@ -99,7 +100,7 @@ public void updateTile(){
 
                 //load up items
                 if(payload.block().hasItems && items.any()){
-                    if(timer(timerLoad, loadTime)){
+                    if(consValid() && timer(timerLoad, loadTime)){
```

Comment:
Instead of checking for consValid(), you should take efficiency into account. 50% power should mean 50% unload speed.

Reply:
did a hacky math, at least it should work.